spec/models/contributor_spec.rb

```diff
@@ -550,19 +550,28 @@
 
     describe 'when many requests are sent to the contributor' do
       before(:each) do
-        another_request = create(:request, broadcasted_at: 1.day.ago)
-        create(:message, request: the_request, recipient: contributor)
-        create(:message, request: another_request, recipient: contributor)
+        previous_request = create(:request, broadcasted_at: 1.day.ago)
+        create(:message, request: the_request, recipient: contributor, created_at: the_request.broadcasted_at)
+        create(:message, request: previous_request, recipient: contributor, created_at: previous_request.broadcasted_at)
       end
 
       it { should eq(the_request) }
     end
 
+    describe 'when most recently a manual message is sent out belonging to a previous request' do
+      let(:previous_request) { create(:request, broadcasted_at: 1.day.ago) }
+      before(:each) do
+        create(:message, request: the_request, recipient: contributor, created_at: the_request.broadcasted_at)
+        create(:message, request: previous_request, recipient: contributor)
+      end
+
+      it { should eq(previous_request) }
+    end
+
```

Comment:
Nice thank you for adding a test for this, it makes it clear what you are trying to achieve.